--- x265/source/CMakeLists.txt~	2015-06-10 08:25:49.866524126 +0200
+++ x265/source/CMakeLists.txt	2015-06-10 08:29:27.775303997 +0200
@@ -153,11 +153,7 @@ if(GCC)
     if(NATIVE_BUILD)
         if(INTEL_CXX)
             add_definitions(-xhost)
-        else()
-            add_definitions(-march=native)
         endif()
-    elseif(X86 AND NOT X64)
-        add_definitions(-march=i686)
     endif()
     if(ARM)
         add_definitions(-march=armv6 -mfloat-abi=hard -mfpu=vfp)
