--- x265_2.9/source/cmake/CMakeASM_NASMInformation.cmake.orig	2018-10-05 14:14:40.000000000 +0200
+++ x265_2.9/source/cmake/CMakeASM_NASMInformation.cmake	2018-10-29 17:47:25.033029663 +0100
@@ -9,7 +9,11 @@
     if(APPLE)
         set(ARGS -f macho64 -DPREFIX)
     elseif(UNIX AND NOT CYGWIN)
+        if("${CMAKE_SIZEOF_VOID_P}" MATCHES 4)
+            set(ARGS -f elfx32)
+        else()
         set(ARGS -f elf64)
+        endif()
     else()
         set(ARGS -f win64)
     endif()
--- x265_2.9/source/CMakeLists.txt.orig	2018-10-29 17:47:51.806362689 +0100
+++ x265_2.9/source/CMakeLists.txt	2018-10-29 17:53:08.726359072 +0100
@@ -50,7 +50,7 @@
     add_definitions(-DX265_ARCH_X86=1)
     if(CMAKE_CXX_FLAGS STREQUAL "-m32")
         message(STATUS "Detected x86 target processor")
-    elseif("${CMAKE_SIZEOF_VOID_P}" MATCHES 8)
+    elseif("${CMAKE_SIZEOF_VOID_P}" MATCHES 8 OR "${LIB_SUFFIX}" STREQUAL "x32")
         set(X64 1)
         add_definitions(-DX86_64=1)
         message(STATUS "Detected x86_64 target processor")
